packages/hoc-input/src/formInput.js

```diff
@@ -73,13 +78,35 @@ const formInput = WrappedComponent => {
         onFocus: this.onFocus,
         onBlur: this.onBlur
       };
+
+      const isSearch = inputProps.type.toLowerCase() === "search";
+      const composedProps = isSearch
+        ? { ...searchProps, ...inputProps }
+        : inputProps;
+      const icon = composedProps.icon && (
+        <div className="icon">
+          <i className={composedProps.icon} />
+        </div>
+      );
+      const clearButton = isSearch &&
+        this.state.value && (
+          <Link className="clear" onClick={this.handleClearClick}>
+            <i className="wg-close-int" />
+          </Link>
+        );
+
       return (
         <div className="input">
+          {icon}
           <WrappedComponent
-            {...inputProps}
+            {...composedProps}
             {...handlers}
             value={this.state.value}
+            ref={input => {
+              this.wrappedComponent = input;
+            }}
```

Comment:
I don't understand why ref is not working.
I'm getting this: `Warning: Stateless function components cannot be given refs. Attempts to access this ref will fail.` but this is not a stateless function.